File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int low = 0;
        int high = nums.size() - 1;
        int n = nums.size();
        while(low<=high){
            int middle = (high+low)/2;
            if(nums[middle]>=target){
                n = middle;
                high = middle - 1;
            }
            else{
                low = middle + 1;
            }
        }
        return n;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        
        s = 0
        e = len(nums) - 1
        
        while(s<=e):
            middle = (s+e)//2
            
            if(target==nums[middle]):
                return middle
            elif(target>nums[middle]):
                s = middle + 1
            elif(target<nums[middle]):
                e = middle - 1
                
                
        return s
        